src/pages/dashboard/nodeSignIn/NodeSignInPSPForm.tsx

```diff
@@ -94,18 +109,24 @@ const NodeSignInPSPForm = ({ goBack, pspNodeData }: Props) => {
     enableReinitialize: true,
   });
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   const submit = async () => {
     setLoading(true);
 
-    if (selectedParty && pspNodeData) {
+    if (selectedParty && isPspSigned(signInData)) {
       try {
-        await updatePSPInfo(formik.values);
+        if (!isPspBrokerSigned(signInData) && intermediaryAvailableValue) {
+          await createPspBroker({
+            broker_psp_code: selectedParty.fiscalCode,
+            description: selectedParty.description,
+            enabled: true,
+            extended_fault_bean: true,
+          });
+        }
 
-        await updateSigninData(selectedParty);
+        await updatePSPInfo(selectedParty.fiscalCode, formik.values);
```

Comment:
Older and imported psp could be signed to the node with other code that not resemble the current party fiscalcode